functests/4_latency/latency.go

```diff
@@ -229,62 +235,74 @@ var _ = Describe("[performance] Latency Test", func() {
+func getLatencyTestRun() (parsedValue bool) {
+	val := false
 	latencyTestRunEnv := os.Getenv("LATENCY_TEST_RUN")
 	if latencyTestRunEnv != "" {
 		if latencyTestRunEnv == "true" {
-			latencyTestRun = true
+			val = true
 		}
 	}
+	return val
 }
 
 //read LATENCY_TEST_RUNTIME environment var and parse it if set
-func getLatencyTestRuntime() {
+func getLatencyTestRuntime() (parsedValue string) {
 	latencyTestRuntimeEnv := os.Getenv("LATENCY_TEST_RUNTIME")
-	if latencyTestRuntimeEnv != "" {
-		latencyTestRuntime = latencyTestRuntimeEnv
-	}
+	return latencyTestRuntimeEnv
 }
 
 //read LATENCY_TEST_DELAY environment var and parse it if set
-func getLatencyTestDelay() {
+func getLatencyTestDelay() (parsedValue int, errMsg error) {
+	var err error
+	var val int
 	latencyTestDelayEnv := os.Getenv("LATENCY_TEST_DELAY")
 	if latencyTestDelayEnv != "" {
-		var err error
-		if latencyTestDelay, err = strconv.Atoi(latencyTestDelayEnv); err != nil {
-			klog.Fatalf("the environment variable LATENCY_TEST_DELAY has incorrect value %q", latencyTestDelayEnv)
+		if val, err = strconv.Atoi(latencyTestDelayEnv); err != nil {
+			err = fmt.Errorf("the environment variable LATENCY_TEST_DELAY has incorrect value %q", latencyTestDelayEnv)
 		}
+
 	}
+	return val, err
 }
 
 //read LATENCY_TEST_CPUS environment var and parse it if set
-func getLatencyTestCpus() {
+func getLatencyTestCpus() (parsedVal int, errMsg error) {
+	var err error
+	var val int
 	latencyTestCpusEnv := os.Getenv("LATENCY_TEST_CPUS")
 	if latencyTestCpusEnv != "" {
-		var err error
-		if latencyTestCpus, err = strconv.Atoi(latencyTestCpusEnv); err != nil {
-			klog.Fatalf("the environment variable LATENCY_TEST_CPUS has incorrect value %q", latencyTestCpusEnv)
+		if val, err = strconv.Atoi(latencyTestCpusEnv); err != nil {
+			err = fmt.Errorf("the environment variable LATENCY_TEST_CPUS has incorrect value %q", latencyTestCpusEnv)
 		}
 	}
+	return val, err
 }
 
-//read MAXIMUM_LATENCY environment var and parse it if set
-func getMaxLatency() {
+// getMaximumLatency should look for one of the following environment variables:
+// OSLAT_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// CYCLICTEST_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// HWLATDETECT_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// MAXIMUM_LATENCY: unified expected maximum latency for all tests
+func getMaximumLatency(testName string) (parsedValue int, errMsg error) {
+	var err error
+	var val int
 	unifiedMaxLatencyEnv := os.Getenv("MAXIMUM_LATENCY")
 	if unifiedMaxLatencyEnv != "" {
 		var err error
-		if maximumLatency, err = strconv.Atoi(unifiedMaxLatencyEnv); err != nil {
-			klog.Fatalf("the environment variable MAXIMUM_LATENCY has incorrect value %q", unifiedMaxLatencyEnv)
+		if val, err = strconv.Atoi(unifiedMaxLatencyEnv); err != nil {
+			err = fmt.Errorf("the environment variable MAXIMUM_LATENCY has incorrect value %q", unifiedMaxLatencyEnv)
```

Comment:
You should return the error here, otherwise, you might override it later, when you're trying to parse the other env variables.

Reply:
correct. In addition you should wrap errors here: https://blog.golang.org/go1.13-errors
TL;DR: use something like
```
err = fmt.Errorf("the environment variable MAXIMUM_LATENCY has incorrect value %q: %w", unifiedMaxLatencyEnv, err)
```

Reply:
fixed, thanks.

Reply:
@fromanirh please keep me honest here, but yet we should have a ```return``` statement here. Storing the error in ```err``` won't keep it from being override later on in the function.

Reply:
it won't indeed

Reply:
After discussing the matter with Talor, I added a return statement on the first error check.Thanks.